client/src/components/Draftail/CommentableEditor/CommentableEditor.js

```diff
@@ -0,0 +1,217 @@
+    const focusedId = useSelector(commentApp.selectors.selectFocused);
+
+    const ids = useMemo(() => comments.map(comment => comment.localId), [comments]);
+
+    const commentStyles = useMemo(() => ids.map(id => {return {
+      type: COMMENT_STYLE_IDENTIFIER + id,
+      style: enabled ? {
+        'background-color': (focusedId !== id) ? '#01afb0' : '#007d7e'
+      } : {}
+    }}), [ids, enabled]);
+
+    const [uniqueStyleId, setUniqueStyleId] = useState(0)
+
+    const previousFocused = usePrevious(focusedId);
+    const previousIds = usePrevious(ids);
+    const previousEnabled = usePrevious(enabled);
+    useEffect(() => {
+      // Only trigger a focus-related rerender if the current focused comment is inside the field, or the previous one was
+      const validFocusChange = previousFocused !== focusedId && (previousIds && previousIds.includes(previousFocused) || ids.includes(focusedId))
+
+      if ((!validFocusChange) && previousIds === ids && previousEnabled === enabled) {
+        return
+      }
+
+      // Filter out any invalid styles - deleted comments, or now unneeded STYLE_RERENDER forcing styles
+      const filteredContent = filterInlineStyles(inlineStyles.map(style => style.type).concat(ids.map(id => COMMENT_STYLE_IDENTIFIER + id)), editorState.getCurrentContent())
+      // Force reset the editor state to ensure redecoration, and apply a new (blank) inline style to force inline style rerender
+      // This must be entirely new for the rerender to trigger, hence the unique style id, as with the undo stack we cannot guarantee
+      // that a previous style won't persist without filtering everywhere, which seems a bit too heavyweight
+      // This hack can be removed when draft-js triggers inline style rerender on props change
+      setEditorState(editorState => forceResetEditorState(editorState, 
+        Modifier.applyInlineStyle(
+          filteredContent,
+          getFullSelectionState(filteredContent),
+          'STYLE_RERENDER_'+uniqueStyleId
+        )
+      ))
+      setUniqueStyleId((id) => (id + 1) % 200);
+    }, [focusedId, enabled, inlineStyles, ids, editorState])
+
+    const timeoutRef = useRef();
+    useEffect(() => {
+      window.clearTimeout(timeoutRef.current);
+      const filteredEditorState = EditorState.push(
+        editorState,
+        filterInlineStyles(inlineStyles.map(style => style.type), editorState.getCurrentContent())
+      );
+      timeoutRef.current = window.setTimeout(
+        onSave(serialiseEditorStateToRaw(filteredEditorState)),
+        250,
+      );
+      return () => {
+        window.clearTimeout(timeoutRef.current);
+      }
+    }, [editorState, inlineStyles]);
+
+    return <DraftailEditor
+    ref={editorRef}
+    onChange={setEditorState}
+    editorState={editorState}
+    controls={enabled ? [CommentControl] : []}
+    decorators={enabled ? [{
+      strategy: findCommentStyleRanges,
+      component: CommentDecorator 
+    }] : []}
```

Comment:
So I should declare it as a variable in the function rather than inline here? Or would I need to memoise it to fix any equality checks by reference failing?

Reply:
My bad, I had missed `CommentDecorator` was declared in this function. Ignore unless there seems to be performance issues! If so, yes, we’d want to memoise.